lib/Core/TerriaError.ts

```diff
@@ -53,69 +99,100 @@ export type TerriaErrorOverrides = Partial<TerriaErrorOptions> | string;
   /** `sender` isn't really used for anything at the moment... */
   readonly sender: unknown;
   readonly originalError?: (TerriaError | Error)[];
-  private _raisedToUser: boolean = false;
+
+  @observable showDetails = false;
 
   /**
    * Convenience function to generate a TerriaError from some unknown error. It will try to extract a meaningful message from whatever object it is given.
    *
-   * If error is a `TerriaError`, then `createParentError` it will be used to create a tree of `TerriaErrors` (see {@link `TerriaError#createParentError}`).
-   *
    * `overrides` can be used to add more context to the TerriaError
+   *
+   * If error is a `TerriaError`, and `overrides` are provided -  then `createParentError` will be used to create a tree of `TerriaErrors` (see {@link `TerriaError#createParentError}`).
+   *
+   * Note, you can not pass `TerriaErrorOptions` (or JSON version of `TerriaError`) as the error parameter.
+   *
+   * For example:
+   *
+   * This is  **incorrect**:
+   *
+   * ```
+   * TerriaError.from({message: "Some message", title: "Some title"})
+   * ```
+   *
+   * Instead you must use TerriaError constructor
+   *
+   * This is **correct**:
+   *
+   * ```
+   * new TerriaError({message: "Some message", title: "Some title"})
+   * ```
    */
   static from(error: unknown, overrides?: TerriaErrorOverrides): TerriaError {
     if (error instanceof TerriaError) {
       return isDefined(overrides) ? error.createParentError(overrides) : error;
     }
 
-    // Try to find message using overrides or error object
-    let message = "Unknown error";
+    // Try to find message from error object
+    let message: string | undefined;
+    // Create original Error from `error` object
+    let originalError: Error | undefined;
 
-    // If overrides is a string - we treat is as the `message` parameter
-    if (typeof overrides === "string") {
-      message = overrides;
-      overrides = {};
-    } else if (typeof error === "string") {
+    if (typeof error === "string") {
       message = error;
+      originalError = new Error(message);
     } else if (error instanceof Error) {
       message = error.message;
+      originalError = error;
     } else if (typeof error === "object" && error !== null) {
       message = error.toString();
+      originalError = new Error(error.toString());
     }
 
     return new TerriaError({
       title: { key: "core.terriaError.defaultTitle" },
-      message,
-      originalError: error instanceof Error ? error : undefined,
-      ...overrides
+      message: message ?? { key: "core.terriaError.defaultMessage" },
+      originalError,
+      ...parseOverrides(overrides)
     });
   }
 
   /** Combine an array of `TerriaErrors` into a single `TerriaError`.
    * `overrides` can be used to add more context to the combined `TerriaError`.
    */
   static combine(
-    errors: TerriaError[],
+    errors: (TerriaError | undefined)[],
```

Comment:
how come we allow undefineds inside the error array?

Reply:
Just for convenience. I think it is only used in one spot

https://github.com/TerriaJS/terriajs/blob/842ffdaae2cff019632e260d173e8fb99f21899d/lib/Core/Result.ts#L100-L115

But I figured, because the return signature has to be `TerriaError | undefined` anyway - i may as well include it in arguments